core/cont/inc/RangeCast.h

```diff
@@ -0,0 +1,185 @@
+// Author: Jonas Rembser, CERN, Jul 2021
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RangeCast_h
+#define ROOT_RangeCast_h
+
+#include <cassert>
+#include <type_traits>
+#include <utility>
+
+namespace ROOT {
+namespace Internal {
+
+#if (__cplusplus < 201700L)
+
+template <typename T, bool isDynamic = false>
+struct Cast {
```

Comment:
Coding convection:
```suggestion
struct Cast_ {
```
or
```suggestion
struct RCast {
```

Reply:
Okay! I used `RCast` now.